relay-spans/src/v2_to_v1.rs

```diff
@@ -202,6 +209,58 @@ fn span_v2_link_to_span_v1_link(link: SpanV2Link) -> SpanLink {
     }
 }
 
+/// Generates a `sentry.op` attribute for V2 span, if possible.
+///
+/// This uses attributes of the span to figure out an appropriate operation name, inferring what the SDK might have sent. Reliably infers an op for well-known OTel span kinds like database operations. Does not infer an op for frontend and mobile spans sent by Sentry SDKs that don't have an OTel equivalent (e.g., resource loads). This relies on the SDK to send a `sentry.op` attribute, since we cannot infer an op.
+fn derive_op_for_v2_span(span: &SpanV2) -> String {
+    // NOTE: `op` is not a required field in the SDK, so the fallback is an empty string.
+    let op = String::from("default");
+
+    let attributes = if let Some(attributes) = span.attributes.value() {
+        attributes
+    } else {
+        return op;
+    };
+
+    let kind = span.kind.value().unwrap_or(&SpanV2Kind::Internal);
+
+    if attributes.get("http.request.method").is_some() || attributes.get("http.method").is_some() {
+        return match kind {
+            SpanV2Kind::Client => String::from("http.client"),
+            SpanV2Kind::Server => String::from("http.server"),
+            _ => {
+                if attributes.get("sentry.http.prefetch").is_some() {
+                    String::from("http.prefetch")
+                } else {
+                    String::from("http")
+                }
+            }
+        };
+    }
+
+    if attributes.get("db.system").is_some() {
+        return String::from("db");
+    }
+
+    if attributes.get("rpc.service").is_some() {
+        return String::from("rpc");
+    }
+
+    if attributes.get("messaging.system").is_some() {
+        return String::from("message");
+    }
+
+    if let Some(faas_trigger) = attributes.get("faas.trigger") {
+        if let Some(trigger_value) = faas_trigger.value() {
+            if let Some(trigger_str) = trigger_value.value.value.value().and_then(|v| v.as_str()) {
+                return trigger_str.to_string();
+            }
+        }
+    }
```

Comment:
Ugh, I think `Attribute` may need an accessor function to make this sort of thing less cumbersome. Even so, it can be made a bit nicer:
```suggestion
    if let Some(faas_trigger) = attributes
        .get("faas.trigger")
        .and_then(|faas_trigger| faas_trigger.value())
        .and_then(|trigger_value| trigger_value.value.value.value())
        .and_then(|v| v.as_str())
    {
        return faas_trigger.to_owned();

```